app/models/season/phase_switcher.rb

```diff
@@ -1,8 +1,11 @@
-module SeasonPhaseSwitcher
-  extend ActiveSupport::Concern
+class Season::PhaseSwitcher
+
+  def initialize(season)
```

Comment:
This is what I meant in my other comment: passing the season as an argument is not really necessary since the switcher can grab the current season from the DB itself.